### ISSUE TITLE
`test --show-events` does not show events from `before` methods

### ISSUE THREAD

**Issue by juharris** [trufflesuite/truffle#2268]
- [ ] I've asked for help in the [Truffle Gitter](http://gitter.im/Consensys/truffle) before filing this issue.

---------------------------

## Issue

`truffle test --show-events` does not show events from `before` methods.

## Steps to Reproduce

Call a contract method that emits events.
When called from a `before` method the events will not be shown.
Change `before` to `it` and the events will show.

## Expected Behavior

`--show-events` should always show the events.

## Actual Results

The events are not shown when the test method is `before`.

## Environment

* Operating System: Windows with WSL Ubuntu 18.04
* Ethereum client: Ganache
* Truffle version (`truffle version`): v5.0.29 (core: 5.0.29)
* node version (`node --version`): v10.9.0
* npm version (`npm --version`): 6.9.0

**Comment by eggplantzzz** [comment]
Hmm interesting, I'm not sure how to feel about this. I can understand the point of view that you might only be concerned about the tests themselves and not the setup. Perhaps we could add some option to include them? What do you think @gnidan and @CruzMolina?

**Comment by juharris** [comment]
I agree that practically you might not want to see events from setting up the test but the docs say "Log **all** contract events." (emphasis mine):
```
truffle help test

  Usage:        truffle test [<test_file>] [--compile-all] [--network <name>] [--verbose-rpc] [--show-events]
  Description:  Run JavaScript and Solidity tests
...
                --show-events
                    Log all contract events.
```

**Comment by eggplantzzz** [comment]
Haha, yeah you are right. Do you feel strongly and/or have a strong argument one way or the other on this issue? I'll get some input from some of the rest of the team.

**Comment by juharris** [comment]
I think it would be nice to for what the docs say and see **all** events even during setup.

**Comment by fainashalts** [comment]
If anyone wants to try to implement this, this should probably be done with a separate flag, like `--show-setup-events` or something similar.

**Comment by juharris** [comment]
Adding another flag is fine but the docs for `--show-events` is still wrong since not all events are shown.  It would be better to make `--show-events` work as explained in the docs.

**Comment by gnidan** [comment]
I'm not sure we want the default behavior for this flag to include events from `before` blocks, since those are used for setup, and likely makes for noise. Regardless, part of the definition of done for this issue should include updating all relevant documentation to be consistent with behavior.

**Comment by Jofairden** [comment]
Ran into this issue today , and using scribble this would be highly useful as the assertion events are not being show when using a before hook.
